src/com/softwaret/kpdf/controller/login/LoginControllerImpl.kt

```diff
@@ -0,0 +1,42 @@
+package com.softwaret.kpdf.controller.login
+
+import ErrorFactory
+import com.softwaret.kpdf.controller.base.BaseController
+import com.softwaret.kpdf.interactor.login.LoginInteractor
+import com.softwaret.kpdf.response.Response
+import com.softwaret.kpdf.response.error.ErrorType
+import com.softwaret.kpdf.response.success.LoginResponse
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Parameters
+
+class LoginControllerImpl(
+    private val loginInteractor: LoginInteractor,
+    errorFactory: ErrorFactory
+) : BaseController(errorFactory), LoginController {
+
+    companion object {
+
+        private const val LOGIN = "login"
+        private const val PASSWORD = "password"
+    }
+
+    override suspend fun login(parameters: Parameters, response: suspend (Response) -> Unit) {
```

Comment:
`response` to słaba nazwa, raczej `respondBlock` albo `onRespond`

Reply:
Z tego bloku bym zrezygnował i jedynie zwracał `Response` w funkcji (łatwiej się testuje)